ui/panels-plugin/src/plugins/pie-chart/pie-chart-model.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright 2024 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ModeOption, SortOption } from '@perses-dev/components';
+import { CalculationType, DEFAULT_CALCULATION, Definition, FormatOptions } from '@perses-dev/core';
+import { OptionsEditorProps, LegendSpecOptions } from '@perses-dev/plugin-system';
+
+import {
+  DEFAULT_CONNECT_NULLS,
+  DEFAULT_POINT_RADIUS,
+  StackOptions,
+  TimeSeriesChartPaletteOptions,
+} from '../time-series-chart/time-series-chart-model';
+
+export const DEFAULT_FORMAT: FormatOptions = { unit: 'decimal', shortValues: true };
+export const DEFAULT_SORT: SortOption = 'desc';
+export const DEFAULT_MODE: ModeOption = 'value';
+export const DEFAULT_LINE_WIDTH = 1.25;
+export const DEFAULT_AREA_OPACITY = 0;
+
+export const VISUAL_CONFIG = {
+  lineWidth: {
+    label: 'Line Width',
+    testId: 'slider-line-width',
+    min: 0.25,
+    max: 3,
+    step: 0.25,
+  },
+  pointRadius: {
+    label: 'Point Radius',
+    testId: 'slider-point-radius',
+    min: 0,
+    max: 6,
+    step: 0.25,
+  },
+  areaOpacity: {
+    label: 'Area Opacity',
+    testId: 'slider-area-opacity',
+    min: 0,
+    max: 1,
+    step: 0.05,
+  },
+  stack: {
+    label: 'Stack Series',
+  },
+  connectNulls: {
+    label: 'Connect Nulls',
+  },
+};
+
+export interface BarChartDefinition extends Definition<PieChartOptions> {
+  kind: 'PieChart';
+}
+export const DEFAULT_VISUAL: TimeSeriesChartVisualOptions = {
+  lineWidth: DEFAULT_LINE_WIDTH,
+  areaOpacity: DEFAULT_AREA_OPACITY,
+  pointRadius: DEFAULT_POINT_RADIUS,
+  connectNulls: DEFAULT_CONNECT_NULLS,
+};
+
+export interface PieChartOptions {
+  legend?: LegendSpecOptions;
+  visual?: TimeSeriesChartVisualOptions;
```

Comment:
Are we sure we need all these options from the time series? Wouldn't be better to include only the options that are relevant for the pie chart?

Reply:
I believe we do need these for piechart as well for displaying the legend(option in the settings). Removed thresholds because that wasn't needed

Reply:
I meant the TimeSeriesChartVisualOptions, this seems out of place as is referencing the time series, but below we are defining the type, so it should be `PieChartChartVisualOptions`

Reply:
made the change to `PieChartChartVisualOptions`